### PR TITLE
feat(commerce): make summary a sub-controller

### PR DESCRIPTION
**Pull request by Spuffynism**
I've also:
- Made it exist as a recs sub-controller
- Adapted the related atomic components
- Renamed `firstSearchExecuted` to `firstRequestExecuted` since `firstSearchExecuted` does not make sense in the context of search and recs.
- Renamed the `ListingSummary` controller type to `ProductListingSummary` to be more in line with how product-listing-related types are named

[CAPI-1044]

[CAPI-1044]: https://coveord.atlassian.net/browse/CAPI-1044?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/atomic/README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ Once you have cloned the repo, follow the instructions in the top-level [README.
 To start the project in development mode, run:
 
 ```bash
-npm run dev
+npm run start
```
Comment (Spuffynism) [inline packages/atomic/README.md:17]: This was out of date as the `dev` script doesn't exist in `package.json` anymore

File: packages/headless/src/controllers/commerce/core/sub-controller/headless-sub-controller.ts [hunk 2]
```diff
@@ -67,10 +75,19 @@ export interface BaseSolutionTypeSubControllers {
    * @returns A `Pagination` sub-controller.
    */
   pagination(props?: PaginationProps): Pagination;
+
+  /**
+   * Builds a `Summary` controller.
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/commerce/core/sub-controller/headless-sub-controller.ts:80]:
```suggestion
   * Creates a `Summary` sub-controller.
```

File: packages/headless/src/controllers/commerce/core/sub-controller/headless-sub-controller.ts [hunk 3]
```diff
@@ -67,10 +75,19 @@ export interface BaseSolutionTypeSubControllers {
...
+   *
+   * @returns A `Summary` controller.
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/commerce/core/sub-controller/headless-sub-controller.ts:82]:
```suggestion
   * @returns A `Summary` sub-controller.
```

File: packages/headless/src/controllers/commerce/core/summary/headless-core-summary.ts [hunk 4]
```diff
@@ -2,109 +2,122 @@ import {CommerceAPIErrorStatusResponse} from '../../../../api/commerce/commerce-
...
-  firstSearchExecuted: boolean;
-
+  firstRequestExecuted: boolean;
   /**
    * The total number of products available.
    */
   totalNumberOfProducts: number;
   /**
-   * Whether the search has returned any products.
+   * Whether the query has returned any products.
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/commerce/core/summary/headless-core-summary.ts:28]:
```suggestion
   * Whether the request has returned any products.
```

File: packages/headless/src/controllers/commerce/core/summary/headless-core-summary.ts [hunk 5]
```diff
@@ -2,109 +2,122 @@ import {CommerceAPIErrorStatusResponse} from '../../../../api/commerce/commerce-
...
    */
   hasProducts: boolean;
   /**
-   * Whether the search is currently loading.
+   * Whether the query is currently loading.
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/commerce/core/summary/headless-core-summary.ts:32]:
```suggestion
   * Whether the request is currently loading.
```

File: packages/headless/src/controllers/commerce/core/summary/headless-core-summary.ts [hunk 6]
```diff
@@ -2,109 +2,122 @@ import {CommerceAPIErrorStatusResponse} from '../../../../api/commerce/commerce-
...
    */
   isLoading: boolean;
   /**
-   * Whether the search has returned an error.
+   * Whether the query has returned an error.
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/commerce/core/summary/headless-core-summary.ts:36]:
```suggestion
   * Whether the request has returned an error.
```

File: packages/headless/src/controllers/commerce/product-listing/sort/headless-product-listing-sort.ts [hunk 7]
```diff
@@ -1,30 +0,0 @@
-import {CommerceEngine} from '../../../../app/commerce-engine/commerce-engine';
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/commerce/product-listing/sort/headless-product-listing-sort.ts:1]: 👏

File: packages/headless/src/controllers/commerce/product-listing/summary/headless-product-listing-summary.ts [hunk 8]
```diff
@@ -1,43 +1,3 @@
...
-  return buildCoreSummary(engine, {
-    options: {
-      selectNumProducts: numberOfProductsSelector,
-      selectResponseId: responseIdSelector,
-      selectLoading: isLoadingSelector,
-      selectError: errorSelector,
-    },
-  });
-};
+export interface ProductListingSummaryState extends SummaryState {}
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/commerce/product-listing/summary/headless-product-listing-summary.ts:3]: Eeeh feels a bit weird to have this in a file of its own, but I can't think of anything that would be better.

**Review by fbeaudoincoveo — COMMENTED** [review]
💯 

This is a breaking change, so we'll have to make sure to communicate it to our early adopters.

File: packages/headless/src/controllers/commerce/core/sub-controller/headless-sub-controller.ts (hunk 2 above)
Comment (Spuffynism) [reply packages/headless/src/controllers/commerce/core/sub-controller/headless-sub-controller.ts]: Applied documentation changes in https://github.com/coveo/ui-kit/pull/4082/commits/a2b6110861ab7c44a5ebfc6a1429aabad825d4e1

**Review by olamothe — APPROVED** [review]

**Review by fbeaudoincoveo — APPROVED** [review]

**Pull request merged**